store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java

```diff
@@ -48,6 +47,9 @@
  */
 public class ExtensionDispatcherServlet extends ZimbraServlet {
     private static Map sHandlers = Collections.synchronizedMap(new HashMap());
+
+    private static Map<String, ArrayList<ExtensionHttpHandler>> EXTENSION_HANDLERS = Collections.synchronizedMap(new HashMap());
```

Comment:
For improved performance in high-concurrency scenarios, it is generally recommended to use `ConcurrentHashMap` over a `synchronizedMap`, as it employs fine-grained locking and provides efficient concurrent access to map data.

Reply:
hmm. that's true, since we are not EXTENSION_HANDLERS as lock here we can go ahead with the ConcurrentHashMap here.